### ISSUE TITLE
Debouncing Google Places Autocomplete Requests

### ISSUE THREAD

**Issue by MysteryDash** [coopcycle/coopcycle-web#946]
Hello,

It is no secret that the Google APIs are expensive, however, with the way CoopCycle currently works, each character added to an address field will generate a call to the Google Places Autocomplete API.

There are two basic solutions to address that matter:
- Make use of the session system of the API (the overall cost should be lower for 10+ characters addresses)
- Debounce the calls to the API.

Here's a sample implementation of the debounce, feel free to change it or use external libraries as needed (that's one of the reasons this is not a pull request).
```javascript
// /js/app/components/AddressAutosuggest.js

// Sample of what a debounce function should look like. 
// Some libraries like jquery-throttle-debounce have similar implementations.
function debounce(func, delay) {
  let inDebounce
  return function() {
    const context = this
    const args = arguments
    clearTimeout(inDebounce)
    inDebounce = setTimeout(() => func.apply(context, args), delay)
  }
}

// The following is a quick & dirty way of testing what the debounce looks like before putting it to use.
this.onSuggestionsFetchRequestedDebounced = debounce(function(value) {
  // @see https://developers.google.com/maps/documentation/javascript/places-autocomplete#place_autocomplete_service
  // @see https://developers.google.com/maps/documentation/javascript/reference/#AutocompleteService
  this.autocompleteService.getPlacePredictions({
    ...autocompleteOptions,
    input: value,
  }, this._autocompleteCallback.bind(this))
}, 500)

onSuggestionsFetchRequested({ value }) {
  this.onSuggestionsFetchRequestedDebounced(value)
}
```

Basically, this works by making the requests to the API only when the user stops typing for a certain amount of time (500ms here).

**Comment by alexsegura** [comment]
![giphy (2)](https://user-images.githubusercontent.com/1162230/70059906-e579a180-15e1-11ea-8ee7-75b397ec866a.gif)

**Comment by alexsegura** [comment]
Thank you for suggesting this! 🙏 
I know about debounce technique but forgot to use it there ☹️
I used `350ms` as the debounce time, based on this [StackOverflow post](https://stackoverflow.com/questions/42361485/how-long-should-you-debounce-text-input) + pifomètre + doigt mouillé
